app/models/invite.rb

```diff
@@ -286,6 +289,20 @@ def ensure_max_redemptions_allowed
       end
     end
   end
+
+  def valid_domain
+    return if self.domain.blank?
+
+    self.domain = domain.downcase
+
+    if self.domain !~ Invite.hostname_regex
+      self.errors.add(:base, I18n.t('invite.invalid_domain', domain: self.domain))
+    end
+  end
+
+  def self.hostname_regex
+    /\A(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])\z/
```

Comment:
Instead of creating this regexp each time the method is called, we can assign this to a constant so that it is only allocated once.